src/main/java/org/opensearch/security/action/apitokens/ApiTokenAction.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.apitokens;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+
+import org.opensearch.client.Client;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.threadpool.ThreadPool;
+
+import static org.opensearch.rest.RestRequest.Method.DELETE;
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.security.action.apitokens.ApiToken.ALLOWED_ACTIONS_FIELD;
+import static org.opensearch.security.action.apitokens.ApiToken.CLUSTER_PERMISSIONS_FIELD;
+import static org.opensearch.security.action.apitokens.ApiToken.CREATION_TIME_FIELD;
+import static org.opensearch.security.action.apitokens.ApiToken.INDEX_PATTERN_FIELD;
+import static org.opensearch.security.action.apitokens.ApiToken.INDEX_PERMISSIONS_FIELD;
+import static org.opensearch.security.action.apitokens.ApiToken.NAME_FIELD;
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class ApiTokenAction extends BaseRestHandler {
+    private final ApiTokenRepository apiTokenRepository;
+
+    private static final List<RestHandler.Route> ROUTES = addRoutesPrefix(
+        ImmutableList.of(
+            new RestHandler.Route(POST, "/apitokens"),
```

Comment:
if we are handling only one token at a time, IMO we should change the endpoint to `apitoken`. Thoughts?

Reply:
I think the standard across the repo is to have it be a plural instead of a singular, even when handling PUT/POST. I think it is just to maintain the same route across all routes of the same action. To follow the standards already set i think plural makes sense here.